Amazon.IonDotnet/Internals/Binary/ManagedBinaryWriter.cs

```diff
@@ -279,6 +270,29 @@ public override void AddTypeAnnotation(string annotation)
             this.userWriter.AddTypeAnnotationSymbol(token);
         }
 
+        /// <inheritdoc />
+        /// <summary>
+        /// This is supposed to close the writer and release all their resources.
+        /// </summary>
+        public override void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (disposing)
```

Comment:
To conform to the pattern, we should probably add a `disposed` flag that causes a quick return. (Here and in other classes where we've implemented the pattern.)

Reply:
Done.